server/lib/src/idm/credupdatesession.rs

```diff
@@ -1899,7 +1907,20 @@ impl IdmServerCredUpdateTransaction<'_> {
         match &session.mfaregstate {
             MfaRegState::TotpInit(totp_token)
             | MfaRegState::TotpTryAgain(totp_token)
+            | MfaRegState::TotpNameTryAgain(totp_token, _)
```

Comment:
Testcase in this file to also assert that the duplicate name gives the correct error etc. 

I think we should potentially also consider limiting *new* labels with https://github.com/kanidm/kanidm/blob/master/server/lib/src/value.rs#L2213

Reply:
Good point. Yeah last night got a bit late so I didn't end up getting to adding tests but I will. Won't get to it tonight but should have it done soon:tm: